src/main/java/ru/hogwarts/school/service/StudentService.java

```diff
@@ -0,0 +1,159 @@
+    }
+    public Long countAllStudents() {
+        logger.info("countAllStudent method has been invoked");
+        return studentRepository.countAllStudents();
+    }
+
+    public Double getAverageAge() {
+        logger.info("getAverageAge method has been invoked");
+        return studentRepository.getAverageAge();
+    }
+
+    public Collection<Student> findFiveLast() {
+        logger.info("findFiveLast method has been invoked");
+        return studentRepository.findFiveLast();
+    }
+
+    public List<String> getAllStartWithA() {
+        return studentRepository.findAll().stream()
+                .map(Student::getName)
+                .map(String::toUpperCase)
+                .filter(s -> s.startsWith("A"))
+                .sorted()
+                .toList();
+    }
+
+    public double getAverageAgeFromStudents() {
+        return studentRepository.findAll().stream()
+                .mapToInt(Student::getAge)
+                .average()
+                .orElse(-1);
+    }
+
+    public int calculate() {
+        long start = System.currentTimeMillis();
+        int result = Stream
+                .iterate(1, a -> a +1)
+                .limit(1_000_000)
+                .parallel()
+                .reduce(0, (a, b) -> a + b );
+        long finish = System.currentTimeMillis();
+        logger.info("Calculate time: " + (finish - start));
+        return result;
+    }
+
+    public void printStudentsNameParallel(){
+       List<Student> students = studentRepository.findAll();
+        System.out.println(students.get(0).getName());
+        System.out.println(students.get(1).getName());
+        System.out.println(students.get(2).getName());
+
+        new Thread(() -> {
+            System.out.println(students.get(3).getName());
+            System.out.println(students.get(4).getName());
+            System.out.println(students.get(5).getName());
+
+        }).start();
+
+        new Thread(() -> {
+            System.out.println(students.get(6).getName());
+            System.out.println(students.get(7).getName());
+            System.out.println(students.get(8).getName());
+
+        }).start();
+
+    }
+
+    public void printStudentsNameSync() {
+        List<Student> students = studentRepository.findAll();
+
+        printStudent(students.get(0).getName());
+        printStudent(students.get(1).getName());
+        printStudent(students.get(2).getName());
+
+            new Thread(() -> {
+                printStudent(students.get(3).getName());
+                printStudent(students.get(4).getName());
+                printStudent(students.get(5).getName());
+
+            }).start();
+
+            new Thread(() -> {
+                printStudent(students.get(6).getName());
+                printStudent(students.get(7).getName());
+                printStudent(students.get(8).getName());
+
+            }).start();
+
+    }
+
+    private void printStudent(String student) {
```

Comment:
```suggestion
    private synchronized void printStudent(String student) {
```
Флаг отдельно не нужен, по факту синхронизируется весь метод, а не какой-то его отдельный кусок